src/mm/memory_manager.rs

```diff
@@ -240,18 +240,24 @@ pub fn is_live_object(object: ObjectReference) -> bool {
     object.is_live()
 }
 
-/// Is the object in the mapped memory?
+/// Is the object in the mapped memory? The runtime can use this function to check
+/// if an object is in MMTk heap.
 ///
 /// Arguments:
 /// * `object`: The object reference to query.
 pub fn is_mapped_object(object: ObjectReference) -> bool {
     object.is_mapped()
 }
 
-/// Is the address in the mapped memory?
+/// Is the address in the mapped memory? The runtime can use this function to check
+/// if an address is mapped by MMTk. Note that this is different than is_mapped_object().
+/// For malloc spaces, MMTk does not map those addresses (malloc does the mmap), so
+/// this function will return false, but is_mapped_object will may return true if the address
+/// is actually a valid object in malloc spaces.
 ///
 /// Arguments:
 /// * `address`: The address to query.
+// TODO: Do we really need this function? Can a runtime always use is_mapped_object()?
```

Comment:
From the difference between the two functions, `is_mapped_object()` and `is_mapped_address()`,
it seems we are exposing some policy-level (e.g. mallocspace) information to the client.

Reply:
Yes, I think we should remove it

Reply:
It seems JikesRVM is using `is_mapped_address()`. We cannot simply remove it.

Reply:
I added a comment to state that the runtime should avoid using this method, as it may get removed in the future. Otherwise than that, I don't know what I can do for it.